### PR TITLE
Get collection document structure

### LINKED ISSUE #11: Feature: Get a collection's document structure
More specifically implement API for: https://www.getoutline.com/developers#tag/Collections/paths/~1collections.documents/post

### PR DESCRIPTION
**Pull request by rsjethani**
fixes #11

### CODE REVIEW

File: collections.go [hunk 1]
```diff
@@ -17,6 +17,11 @@ func newCollectionsClient(sl *sling.Sling) *CollectionsClient {
 	return &CollectionsClient{sl: sl}
 }
 
+// Structure gives access to id's document structure.
```
Comment (StefMa) [inline collections.go:20]: With the `Collections.Create` and `Documents.Create` we started to add the API reference.
See [this line](https://github.com/ioki-mobility/go-outline/pull/35/files#diff-0ba7b976e3f9b89e648e3ba228386a911e97656963fb62cf41ff46697c0bf79cL29).

Do we want to continue with that? 🤔

File: collections.go [hunk 2]
```diff
@@ -17,6 +17,11 @@ func newCollectionsClient(sl *sling.Sling) *CollectionsClient {
...
+func (cl *CollectionsClient) Structure(id CollectionID) *CollectionsStructureClient {
```
Comment (StefMa) [inline collections.go:21]: What about renaming it to
```suggestion
func (cl *CollectionsClient) DocumentStructure(id CollectionID) *CollectionsDocumentStructureClient {
```
? And all the other related functions and structs as well? 🤔

File: internal/cli/cli.go [hunk 3]
```diff
@@ -66,14 +69,49 @@ func collectionCmdFetch(rootCmd *cobra.Command) *cobra.Command {
...
+
 			client := outline.New(baseUrl, &http.Client{}, apiKey)
 			for _, colId := range args {
 				col, err := client.Collections().Get(outline.CollectionID(colId)).Do(context.Background())
 				if err != nil {
 					return fmt.Errorf("can't get collection with id '%s': %w", colId, err)
 				}
 
-				b, err := json.Marshal(col)
+				b, err := json.MarshalIndent(col, "", "  ")
```
Comment (StefMa) [inline internal/cli/cli.go:114]: Please also change this in the `collectionCmdCreate` function

File: internal/cli/cli.go [hunk 4]
```diff
@@ -66,14 +69,49 @@ func collectionCmdFetch(rootCmd *cobra.Command) *cobra.Command {
...
+		Args:  cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			baseUrl, err := rootCmd.Flags().GetString(flagNameBaseUrl)
+			if err != nil {
+				return fmt.Errorf("required flag '%s' not set: %w", flagNameBaseUrl, err)
+			}
+			apiKey, err := rootCmd.Flags().GetString(flagNameApiKey)
+			if err != nil {
+				return fmt.Errorf("required flag '%s' not set: %w", flagNameApiKey, err)
+			}
```
Comment (StefMa) [inline internal/cli/cli.go:105]: We dublicate this already three times.
Maybe it make sense to introduce a function like
```go
func requiredFlags(rootCmd *cobra.Command) (baseUrl string, apiKey string, err error) {
...
}
```

An alternative would be to move the `outline` client creation outside of these sub command to its "parent" command (the `collection` command).
Then we check the required flag only once. If successfully, we can create the `outline` client and put it as an argument to the sub-commands...

What do you think? 🤔

File: collections.go [hunk 5]
```diff
@@ -31,6 +36,40 @@ func (cl *CollectionsClient) Create(name string) *CollectionsCreateClient {
...
+		ID CollectionID `json:"id"`
+	}{ID: id}
+
+	copy := sl.New()
+	copy.Post(common.CollectionsStructureEndpoint()).BodyJSON(&data)
+
+	return &CollectionsStructureClient{sl: copy}
+}
+
+type DocumentStructure []DocumentSummary
```
Comment (StefMa) [inline collections.go:54]: Is this a typicall Go pattern? 🤔 
Why not simply returning a `[]DocumentSummary`? 🤔

**Review by StefMa — CHANGES_REQUESTED** [review]
Looks good overall 👍 
Just a few nitpicking comments...

File: collections.go (hunk 1 above)
Comment (rsjethani) [reply collections.go:20]: Well, the objective of our package is to be intuitive enough to does not feel the need to look at the actual HTTP API. Ideally we only mention the API once in say README and that is it. But sure will add it, philosophical things can be polished later.

File: collections.go (hunk 2 above)
Comment (rsjethani) [reply collections.go:21]: sure :+1:

File: collections.go (hunk 5 above)
Comment (rsjethani) [reply collections.go:54]: Well it is more case-by-case basis. Here we do it to be more in line with other API methods that is returning a single type representing the whole model and not a list/map of types.

File: internal/cli/cli.go (hunk 4 above)
Comment (rsjethani) [reply internal/cli/cli.go:105]: Yes, but doing this properly requires enough refactoring to make it worthy of a separate task.

**Review by StefMa — APPROVED** [review]
:shipit:

**Pull request merged**